flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolPendingRequestFailureTest.java

```diff
@@ -99,6 +105,40 @@ public void testFailingAllocationFailsPendingSlotRequests() throws Exception {
 		}
 	}
 
+	@Test
+	public void testFailingAllocationFailsRemappedPendingSlotRequests() throws Exception {
+		final List<AllocationID> allocations = new ArrayList<>();
+		resourceManagerGateway.setRequestSlotConsumer(slotRequest -> allocations.add(slotRequest.getAllocationId()));
+
+		try (SlotPoolImpl slotPool = setUpSlotPool()) {
```

Comment:
Could we reuse/deduplicate `setUpSlotPool` also in `SlotPoolImplTest`?

Reply:
It's possible but unrelated. I do not feel like we should cleanup the tests here. 
Like you said, the scheduler might also be not needed. I think it can be a larger task and would be better to be a separate task to fully rework the `SlotPoolImpl` tests, including those in `SlotPoolImplTest`, `SlotPoolInteractionsTest`, `SlotPoolSlotSharingTest`.

Reply:
Possibly these test classes can share the same test base.

Reply:
I can open a task for it if you feel it is Ok.

Reply:
sure, I agree that  we can do the overall test cleanup as a separate issue if it is too much for this PR.

Reply:
FLINK-18355 is opened for simplify the tests.